Doc/library/pathlib.rst

```diff
@@ -159,7 +159,7 @@ we also call *flavours*:
 
       class MyPath(PurePosixPath):
           def __init__(self, *pathsegments, template=None, session_id=None):
-              super().__init__(*pathsegments)
+              super().__init__(*pathsegments, template=template)
```

Comment:
Should we add a test to make sure diamond inheritance works?

Reply:
I'm not sure. There's only one place in `pathlib.py` where we call `super()`, and that only exists because we need to raise a deprecation warning when additional arguments are supplied to `pathlib.Path()`. The `Path.__init__()` method will be removed in 3.14, at which point it will be impossible for the test to fail.

Reply:
Perhaps a hidden `.. doctest::` block would be best?